R/nullModelApproxSE.R

```diff
@@ -0,0 +1,217 @@
+## estimate the ratio (se.ratio) of the true SE(score) to the fast SE(score)
+## update the null model to contain this parameter
+## this needs to be run before using the approx.score.se option in assocTestSingle
+## this is based off of the SAIGE variance approximation
+
+
+setGeneric("fitNullModelApproxSE", function(x, ...) standardGeneric("fitNullModelApproxSE"))
+
+setMethod("fitNullModelApproxSE",
+          "SeqVarData",
+          function(x, 
+                   gdsobj,
+                   outcome,
+                   covars = NULL,
+                   cov.mat = NULL,
+                   group.var = NULL,
+                   family = "gaussian",
+                   start = NULL,
+                   AIREML.tol = 1e-4,
+                   max.iter = 100,
+                   EM.iter = 0,
+                   drop.zeros = TRUE,
+                   return.small = FALSE,
+                   nvar = 100, 
+                   min.mac = 20, 
+                   variant.id = NULL, 
+                   sparse = TRUE, 
+                   imputed = FALSE, 
+                   male.diploid=TRUE, 
+                   genome.build=c("hg19", "hg38"), 
+                   verbose=TRUE) {
+
+               # fit the null model
+               null.model <- fitNullModel(sampleData(x), ...)
+               # calculate score.SE and the fast approximation
+               tab <- calcScoreSE(gdsobj, null.model, ...)
+               # update the null model with the se.correction factor
+               null.model <- updateNullModApproxSE(null.model, tab)
+               return(null.model)
+          })
+
+
```

Comment:
Could also have this function return the `Score` value with minimal extra computation, and just call it `calcScore`. I wonder if this would have general utility...